inc/JpegMeta.php

```diff
@@ -3109,6 +3121,10 @@ function & _getFixedString(&$data, $pos, $length = -1) {
         }
 
         $rv = substr($data, $pos, $length);
+        if (strlen($rv) != $length) {
+            trigger_error('Damaged image: '.$this->_fileName, E_USER_WARNING);
```

Comment:
Do you have test image to try this with? I'm not sure using trigger_error is the best option here. In general I would prefer an exception, but I understand that you want to continue working with the image and ignore the problem. Maybe simply logging the error would be enough?

Reply:
I'll try to implement it so that it either logs just the name, or throws an exception that would make it stop parsing metadata. Also it turns out I'll need to run this check for other methods, like _getShort, since those also throw warnings.